### PR TITLE
kaneko16 : Cleanup OKI Bankswitching

### PR DESCRIPTION
**Pull request by cam900**

### CODE REVIEW

File: src/mame/drivers/kaneko16.cpp [hunk 1]
```diff
@@ -170,40 +180,32 @@ WRITE16_MEMBER(kaneko16_state::kaneko16_soundlatch_w)
 }
 
 /* Two identically mapped YM2149 chips */
-
-READ16_MEMBER(kaneko16_state::kaneko16_ay1_YM2149_r)
+template<int chip>
```
Comment (cuavas) [inline src/mame/drivers/kaneko16.cpp]: Please use `CamelCase` for template parameters to make it clear they aren't locals or function parameters.

File: src/mame/drivers/kaneko16.cpp [hunk 2]
```diff
@@ -160,6 +160,16 @@ WRITE16_MEMBER(kaneko16_state::kaneko16_coin_lockout_w)
 
 ***************************************************************************/
 
+void kaneko16_state::kaneko16_common_oki_bank_w(int bankno, int bank, size_t fixedsize, size_t bankedsize)
+{
+	size_t length = (m_okiregion[bankno]->bytes() - fixedsize) / bankedsize;
+
+	if (bank < length)
+	{
+		m_okibank[bankno]->set_enrty(bank);
```
Comment (ajrhacker) [inline src/mame/drivers/kaneko16.cpp]: You should at least be trying to compile your own changes before submitting pull requests filled with obvious compilation errors like this typo.

**Comment by cuavas** [comment]
Please address the issues with your existing pull requests before opening even more of them, and please test your changes before opening pull requests.

**Comment by Osso13** [comment]
This PR needs to be updated to latest APIs. I assume most earlier open PRs will need it too.

**Pull request merged**